libtenzir/builtins/operators/compress_decompress.cpp

```diff
@@ -117,14 +117,38 @@ struct input_buffer {
 struct operator_args {
   located<std::string> type = {};
   std::optional<located<int>> level = {};
-  // TODO: gzip has some further options, which we should also cover.
+  // used by gzip compress
+  std::optional<located<std::string>> gzip_format = {};
+  // used by gzip & brotli compress
+  std::optional<located<int>> window_bits = {};
+
+  auto gzip_format_enum() const -> arrow::util::GZipFormat {
+    if (not gzip_format) {
+      TENZIR_ASSERT(arrow::util::GZipCodecOptions{}.gzip_format
+                    == arrow::util::GZipFormat::GZIP);
+      return arrow::util::GZipFormat::GZIP;
+    }
+    if (gzip_format->inner == "zlib") {
+      return arrow::util::GZipFormat::ZLIB;
+    }
+    if (gzip_format->inner == "deflate") {
+      return arrow::util::GZipFormat::DEFLATE;
+    }
+    if (gzip_format->inner == "gzip") {
+      return arrow::util::GZipFormat::GZIP;
+    }
+    TENZIR_UNREACHABLE();
+  }
```

Comment:
You don't have to change this now, but FWIW, this code would benefit from "parse-don't-validate".